tests/network/primary_pod_network.go

```diff
@@ -0,0 +1,146 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Primary Pod Network", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred(), "Should successfully initialize an API client")
+	})
+
+	Describe("Status", func() {
+		AssertReportedIP := func(vmi *v1.VirtualMachineInstance) {
+			By("Getting pod of the VMI")
+			vmiPod := tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault)
+
+			By("Making sure IP reported on the VMI matches the one on the pod")
+			Expect(vmi.Status.Interfaces[0].IP).To(Equal(vmiPod.Status.PodIP))
+			Expect(vmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+			Expect(vmi.Status.Interfaces[0].IPs[0]).To(Equal(vmiPod.Status.PodIP))
+		}
+
+		Context("VMI connected to the pod network using the default (implicit) binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithDefaultBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using bridge binding", func() {
+			var vmi *v1.VirtualMachineInstance
+
+			BeforeEach(func() {
+				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
+			})
+
+			AfterEach(func() {
+				cleanupVMI(virtClient, vmi)
+			})
+
+			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
+		})
+
+		Context("VMI connected to the pod network using masquerade binding", func() {
```

Comment:
What is the reason for not using `table.DescribeTable` and passing the vmi as a parameter?

Reply:
There is a need to parametrize the fixtures (BeforeEach/AfterEach), the table would not provide that.

Reply:
Since tables are all based on the same test body, having a `BeforeEach` with parameters adds pretty much no value to directly adding the `BeforeEach` body to the test body itself. At the end you have only one test-body. Maybe it would be convenient in some cases or slightly nicer, but at the end the difference to directly doing the setup in the body makes no difference. This is of course different when we are talking about `Contexts` and It, where `BeforeEach` adds much more value. If you think this is a missing feature to allow you using tables, I would suggest to add an Issue on the Ginkgo project and maybe implement it also. I don't  think that missing parametrizable `BeforeEach` blocks which are explicitly meant for a table are a blocker for the use of table driven tests.

Reply:
In this case the `Entry` could just have accepted `vmiWithMsqueradeBinding` and `vmiWithBridgeBinding`. The afterEach would not be necessary if the test writers of networking tests would not insist of sharing infrastructure between tests. While this may make some tests execute faster, it makes it hard to let the networking tests benefit from general test suite improvements.